### DISCUSSION TITLE
Node: use tsx or transpile to js?

### DISCUSSION THREAD

**Discussion by bartoszkrawczyk2** [honojs/hono#1388] [Q&A]
Hi!

Should I use `tsx` command to run the app in production? Are there any performance downgrades? Or should I transpile to js (eg. via `esbuild --bundle`) and run with `node` command?

Thanks!

**Comment by yusukebe** [comment]
Hi @bartoszkrawczyk2,

The reason we use `tsx` in the starter is due to its utility. While `tsx` employs `esbuild` internally, there might be a slight overhead during the initial compilation. For optimization, consider bundling it with `esbuild` into plain JS and then executing it using the `node` command.
